### PR TITLE
[SYCLomatic] Migrate CUDART_VERSION and __CUDART_API_VERSION to DPCT_COMPAT_RT_VERSION

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/test/dpct/predefined_macro_replacement.cu [hunk 1]
```diff
@@ -123,7 +126,7 @@ int main() {
 #endif
 
 int foo(int num) {
-//CHECK: #if (SYCL_LANGUAGE_VERSION >= 202000)
+//CHECK: #if DPCT_COMPAT_RT_VERSION >= 4000
```
Comment (zhimingwang36) [inline clang/test/dpct/predefined_macro_replacement.cu:129]: which file define this macro?

File: clang/test/dpct/predefined_macro_replacement.cu [hunk 2]
```diff
@@ -1,5 +1,8 @@
 // RUN: dpct --format-range=none --usm-level=none -out-root %T/predefined_macro_replacement %s --cuda-include-path="%cuda-path/include" -- -x cuda --cuda-host-only -D__NVCC__ -D__CUDACC__
 // RUN: FileCheck --input-file %T/predefined_macro_replacement/predefined_macro_replacement.dp.cpp --match-full-lines %s
+
+//CHECK: #define DPCT_COMPAT_RT_VERSION {{[1-9][0-9]+}}
```
Comment (zhimingwang36) [inline clang/test/dpct/predefined_macro_replacement.cu:4]: the value of the macro is defined accord to the header file used during migration?

Comment (zhiweij1) [reply clang/test/dpct/predefined_macro_replacement.cu:4]: Yes, it depends on which version header file used.

**Review by ziranzha — APPROVED** [review]

**Review by ziranzha — APPROVED** [review]

**Pull request merged**